### PR TITLE
IO editor: filter payload based on WDL IO, not existing config's IO [GAWB-2775]

### PR DESCRIPTION
**Pull request by aednichols**
Fixes a bug where a user could add an input to their WDL, it would show up in the IO editor, but anything they entered wasn't being saved (which only becomes evident after a refresh).

Should address GAWB-2775 without breaking GAWB-2658 and GAWB-2673. Reviewer, please test that hypothesis, as I do not fully trust my own understanding.

- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: If you changed a URL that is used elsewhere (e.g. in an email), comment about where it is used and ensure the dependent code is updated.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [ ] **Submitter**: If you're adding new automated UI tests, review the test plan with QA
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Product Owner** sign off
- [ ] **Submitter**: Verify all tests go green, including CI tests and automated UI tests.
- [ ] **Submitter**: Squash commits and merge to develop. If adding test code, merge application code and test code at the same time.
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

**Comment by davidangb** [comment]
This looks like the right solution. Design-wise, the `select-keys` filter should work against the WDL IO.

Could this be automated? (is it already?)

File: src/cljs/main/broadfcui/common/method/config_io.cljs [hunk 1]
```diff
@@ -30,8 +30,8 @@
    :save
    (fn [{:keys [props state locals]}]
      (swap! state dissoc :editing?)
-     {:inputs (select-keys (:inputs @locals) (-> props :values :inputs keys))
-      :outputs (select-keys (:outputs @locals) (-> props :values :outputs keys))})
+     {:inputs (select-keys (:inputs @locals) (map #(keyword (:name %)) (:inputs (:inputs-outputs props))))
+      :outputs (select-keys (:outputs @locals) (map #(keyword (:name %)) (:outputs (:inputs-outputs props))))})
```
Comment (MattPutnam) [inline src/cljs/main/broadfcui/common/method/config_io.cljs]: `#(keyword (:name %))` can also be written as `(comp keyword :name)`

**Comment by aednichols** [comment]
@davidangb have been looking into a selenium for this with Brian. He helped me un-break my local setup; it seems that the tests for this area are ignored due to flakiness

**Comment by aednichols** [comment]
I estimate 1 point or so to add a test because I'm a noob at selenium, and the repro case is complicated.

**Comment by aednichols** [comment]
jenkins retest

**Comment by aednichols** [comment]
jenkins retest

**Comment by aednichols** [comment]
jenkins retest

**Comment by aednichols** [comment]
Test update: the tests that failed in the last run aren't related to this change, re-running yet again.

Test for this feature still in progress.

**Comment by aednichols** [comment]
jenkins retest

**Comment by aednichols** [comment]
jenkins retest

**Review by MattPutnam — APPROVED** [review]
👍

**Review by davidangb — APPROVED** [review]
:+1: for code; waiting on tests

**Pull request merged**